tests/unittest.py

```diff
@@ -97,7 +100,7 @@ def value(self) -> _ExcType:
         ...
 
 
-def around(target):
+def around(target: TV) -> Callable[[Callable[P, R]], None]:
```

Comment:
Unfortunately I don't think mypy can use this signature to determine the type of a function decorated by `@around(self)`. (Try reveal_type() on such a function).

We only seem to use this for modifying `setUp` and `tearDown` in our own TestCase subclass. Bluntly, I think this would be less surprising if expressed with inheritance rather than 2014 lisp-in-Python shenanigans.

But I don't think we ought to change any of that here and now.

Reply:
> We only seem to use this for modifying `setUp` and `tearDown` in our own TestCase subclass. Bluntly, I think this would be less surprising if expressed with inheritance rather than 2014 lisp-in-Python shenanigans.

Yeah, definitely.

> But I don't think we ought to change any of that here and now.

I came to the same conclusion that I didn't feel like messing with this here.